src/Workspaces/Core/Portable/Diagnostics/SkippedHostAnalyzersInfo.cs

```diff
@@ -135,6 +145,28 @@ namespace Microsoft.CodeAnalysis.Diagnostics
                     }
                 }
 
+                if (hostAnalyzer is DiagnosticSuppressor suppressor)
+                {
+                    var suppressionDescriptors = suppressor.SupportedSuppressions;
+                    foreach (var descriptor in suppressionDescriptors)
+                    {
+                        if (projectAnalyzerDiagnosticIds.Contains(descriptor.SuppressedDiagnosticId))
+                        {
+                            // Host analyzer cannot suppress nuget analyzers diagnostics.
+                            skippedDiagnosticIdsBuilder.Add(descriptor.Id);
+                        }
+                        else if (projectSuppressedDiagnosticIds.Contains(descriptor.SuppressedDiagnosticId))
+                        {
+                            // Host analyzer lets the nuget installed analyzer suppress the diagnostic if there is overlap.
+                            skippedDiagnosticIdsBuilder.Add(descriptor.Id);
```

Comment:
shouldn't we add `descriptor.SuppressedDiagnosticId` here?

Reply:
No, the current implementation is correct. We should not skip suppressed diagnostic ID here.

Reply:
Lets just return `false` here instead of adding to `skippedDiagnosticIdsBuilder` and keep it simple.

Reply:
@mavasani  This goes to your request to make it if there is any overlap, then turn off the host suppressor correct?  Comment below.

Reply:
Yep.

Reply:
Should be fixed.